src/rely/AssertionState.re

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */;
+type assertionExpectation =
+  | HasAssertions(option(Printexc.location), string)
+  | NoExpectation
+  | HasNAssertions(int, option(Printexc.location), string);
+
+type assertionResult =
+  | Passed
+  | Failed(string, option(Printexc.location), string);
+
+type t = {
+  expectation: assertionExpectation,
+  assertions: list(assertionResult),
```

Comment:
Given we only use this for counting right now/potential memory issues, I could see just making this a number that we increment.